### ISSUE TITLE
aboutKMaxPooling function

### ISSUE THREAD

**Issue by mymuli** [rwth-i6/fgvc#1]
Great job!
https://github.com/rwth-i6/fgvc/blob/ab0db1e5e9f4b4a779d3fa961fa31a4823563c23/elope_torch/KMaxPooling.lua#L19
I would like to ask if there is a Python version of the **KMaxPooling** function?

**Comment by mymuli** [comment]
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Sat Sep 11 11:25:46 2021

@author: muli
"""
import torch
from torch import nn


class TopKPool(nn.Module):
    def __init__(self, k):
        super(TopKPool, self).__init__()
        self.k = k

    def forward(self, x):
        print("x: ", x)
        b, c, _, _ = x.shape
        x = x.view(b, c, -1)
        # 在h*w面上取topK大值
        topkv, _ = x.topk(self.k, dim= -1)
        # 对topK个值取平均
        out = topkv.mean(dim= -1)
        return out


if __name__ == "__main__": 
    aa = torch.randn(2, 3, 5, 5)
    print("aa: ", aa.size())
    print("----------------------------")
    print("aa: ", aa)
    # 最大池化
    bb = nn.MaxPool2d(kernel_size=5, stride=2)
    c = bb(aa)
    print("c: ", c)
    print("c.shape", c.shape)

    print("----------------------------")
    # topK-maxpool
    topkmaxpool = TopKPool(k=2)
    dd = topkmaxpool(aa)
    print("dd: ", dd)
    print("dd.shape", dd.shape)

**Comment by hhanselmann** [comment]
Here is a simple implementation for pytorch:

```
import torch

class KMaxPooling(torch.nn.Module):

    def __init__(self, k, is_global):
        super(KMaxPooling, self).__init__()
        self.k = k
        self.is_global = is_global

    def forward(self, data):
        B = data.shape[0]
        D = data.shape[1]
        H = data.shape[2]
        W = data.shape[3]

        x = data.view(B, D, H*W)
        topk = x.topk(self.k, 2, sorted=True)
        topKActivations = topk[0]

        if self.is_global:
            output = topKActivations.sum(2)
            output = output / self.k
        else:
            output = topKActivations
            
        return output
```

Hope this helps.
